istio-proxy/fetch.sh

```diff
@@ -2,28 +2,56 @@ set -x
 set -e
 
 PROXY_DIR=${PROXY_DIR:-istio-proxy}
+
 function check_envs() {
   if [ -z "$FETCH_DIR" ]; then
     echo "FETCH_DIR required. Please set"
     exit 1
   fi
 
-	PROXY_FETCH_DIR=${FETCH_DIR}/${PROXY_DIR}
+  PROXY_FETCH_DIR=${FETCH_DIR}/${PROXY_DIR}
   CACHE_DIR=${PROXY_FETCH_DIR}/bazel
 }
 
+function check_git_hashes() {
+  local commit_hash_name="$1"
+  local tarball_hash_name="$2"
+  eval commit_hash=\$$commit_hash_name
```

Comment:
Any reason actual hash values cannot be passed in?

Reply:
The caller of fetch does pass in the actual hash and tarball sha256. The $$commit_hash_name and \$$tarball_hash_name are so I can use the same function for different hash names (e.g. PROXY_GIT_COMMIT_HASH or ENVOY_OPENSSL_GIT_COMMIT_HASH). I'm not passing in the value - I'm passing in the env variable name.